pyobjus/pyobjus_conversions.pxi

```diff
@@ -684,31 +708,34 @@ cdef void* convert_py_arg_to_cy(arg, sig, by_value, size_t size) except *:
                 (<void**>val_ptr)[0] = <void*><unsigned long long>arg
             elif type(arg) is str:
                 # passing bytes as void* is the same as for char*
-                (<char **>val_ptr)[0] = <char *><bytes>arg
+                b_arg = <bytes>arg.encode("utf8")
+                Py_INCREF(b_arg)
+                (<char **>val_ptr)[0] = <char *>b_arg
             else:
                 if type(arg) is float:
                     (<float*>arg_val_ptr)[0] = <float>arg
                 elif type(arg) is int:
                     (<int*>arg_val_ptr)[0] = <int>arg
                 (<void**>val_ptr)[0] = <void*>arg_val_ptr
-        
+
     # TODO: method is accepting bit field
-    elif sig[0] == 'b':
+    elif sig.startswith(b'b'):
         raise ObjcException("Bit fields aren't supported in pyobjus!")
 
     # method is accepting unknown type (^?)
-    elif sig[0] == '?':
+    elif sig.startswith(b'?'):
         if by_value:
             assert(0)
         else:
             (<void**>val_ptr)[0] = arg_val_ptr
 
     else:
+        dprint("WARNING: Unknown signature component, passing nil")
         (<int*>val_ptr)[0] = 0
 
     # TODO: Find best time to dealloc memory used by this pointer
-    if arg_val_ptr != NULL and del_arg_val_ptr:
-        free(arg_val_ptr)
-        arg_val_ptr = NULL
+    # if arg_val_ptr != NULL and del_arg_val_ptr:
+    #     free(arg_val_ptr)
+    #     arg_val_ptr = NULL
```

Comment:
let's delete these lines rather than commenting them, what do you think?

Reply:
no because it will need to be fixed at some point like the segfault, i've a feeling it come from this.

Reply:
then it probably deserves a comment/ticket to tell other maintainers/reviewers about it :wink: